### PR TITLE
Adding the types of parameters and returns of functions

### PR DESCRIPTION
**Pull request by stefanycoimbra**
Line 8: Add types of Union and List to use later

Line 24: 
function plot_histograms():
profiler: StructuredProfiler, 
column_names: List[Union[int,str]] = None, 
column_inds: List[int] = None,
return type -> plt.figure

Line 78:
function is_index_graphable_column(): return type -> bool

Line 137:
function plot_col_histogram():
data_type_profiler: Union[int, float], 
ax: matplotlib.axes.Axes=None, 
title: str=None,
return type -> matplotlib.axes

Line 171:
function plot_missing_values_matrix():
profiler: StructuredProfiler, 
ax: matplotlib.axes.Axes=None, title: str=None,
return type -> matplotlib.figure

Line 191:
function plot_col_missing_values():
col_profiler_list: List[StructuredColProfiler], 
ax: matplotlib.axes.Axes=None, 
title: str=None,
return type -> matplotlib.figure

### CODE REVIEW

**Comment by taylorfturner** [comment]
@stefanycoimbra looks like tests are failing on `TestGraphImport.test_no_matplotlib`

File: dataprofiler/reports/graphs.py
Comment (taylorfturner) [inline dataprofiler/reports/graphs.py:24]: instead of `plt.figure` I'd recommend `matplotlib.pyplot.figure` if possible for specificity and so it isn't requiring a particular way of importing matplotlib.

this might also be an opportunity for `from __futures__ import annotations`

**Comment by taylorfturner** [comment]
related to issue #607

**Comment by taylorfturner** [comment]
@stefanycoimbra just checking back in on this. Let us know if you have any questions

**Comment by taylorfturner** [comment]
@stefanycoimbra you'll need to `update branch` too after a fix I just merged in to `main`

**Comment by taylorfturner** [comment]
@stefanycoimbra just checking in. Any questions? Thanks!

File: dataprofiler/reports/graphs.py
Comment (taylorfturner) [inline dataprofiler/reports/graphs.py:73]: Technically this is getting cast as a `List[Any]`. Casting to `List[Union[str, list]]` doesn't work because `_col_name_to_idx` in `StructuredProfiler` is cast to `List[any]` as well. So if we cast to `List[Union[str, int]]`, we can get an error saying the `List[Any]` and `List[Union[str, int]]` are not compatible.

File: dataprofiler/reports/graphs.py
Comment (taylorfturner) [inline dataprofiler/reports/graphs.py:10]: removed for only one import for code and typing usage

File: dataprofiler/reports/graphs.py
Comment (taylorfturner) [inline dataprofiler/reports/graphs.py:11]: required for typing on `plot_col_histogram`

File: dataprofiler/reports/utils.py
Comment (taylorfturner) [inline dataprofiler/reports/utils.py:55]: `require_module` update required to remove `#type: ignore` in `graphs.py`

File: dataprofiler/reports/graphs.py
Comment (taylorfturner) [reply dataprofiler/reports/graphs.py:72]: This should match better but at the same time want to keep the scope smaller for this PR.

**Review by micdavis — COMMENTED** [review]
Will approve once the list[any] comment above is addressed. https://github.com/capitalone/DataProfiler/pull/681#discussion_r997402413

**Comment by taylorfturner** [comment]
> Will approve once the list[any] comment above is addressed. [#681 (comment)](https://github.com/capitalone/DataProfiler/pull/681#discussion_r997402413)

Yes, it should be ... I recommend we keep the scope smaller on the PR and note this in an issue IMO

**Review by micdavis — DISMISSED** [review]

File: dataprofiler/reports/graphs.py
Comment (JGSweets) [reply dataprofiler/reports/graphs.py:73]: Does fixing the one line in the other file fix it though?

Comment (taylorfturner) [reply dataprofiler/reports/graphs.py:73]: unfortunately, it didn't work when I tested that locally

File: dataprofiler/reports/graphs.py
Comment (JGSweets) [inline dataprofiler/reports/graphs.py:33]: optional

Comment (taylorfturner) [reply dataprofiler/reports/graphs.py]: added

**Review by micdavis — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

**Comment by taylorfturner** [comment]
Thanks for getting started on this work, @stefanycoimbra! Great to see all the static typing coming together 🥳

**Pull request merged**
